nodl_python/nodl/errors.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Optional
+
+from lxml import etree
+
+
+class NoDLError(Exception):
+    """Base class for all NoDL exceptions."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+class InvalidNoDLError(NoDLError):
+    """Exception class representing most errors in parsing the NoDL tree."""
+
+    def __init__(self, message: str, invalid: Optional[etree.DocumentInvalid] = None) -> None:
```

Comment:
The functionality contained within this class is not utilized by most subclasses. I think it makes sense to have this class be a parent to all the parsing issues, but consider moving the handling of `invalid` to another child class instead. This base class could be literally empty and it would still serve a good purpose.

Reply:
36f9562